streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -173,7 +173,7 @@ Cancellable schedule(final Duration interval,
      * @param key key
      * @param value value
      */
-    <K, V> void forward(final K key, final V value);
+    <K1 extends K, V1 extends V> void forward(final K1 key, final V1 value);
```

Comment:
This is a "compatibility trick". It lets us bound calls to "forward" when `K` and `V` are specified, and it doesn't affect the type bound at all if people continue using the raw type.

The algebra is that if you have a raw reference, then K and V are both bound to `Object`, in which case `<K1 extends K>` = `<K1 extends Object>` = `<K1>` (that last step is an identity in the type system). So, when there's a raw type, we wind up with the same `<K1, V1>` that we had before (although they were named `<K, V>` here before).